### PR TITLE
feat: add dataextraction functions

### PR DESCRIPTION
**Pull request by laurasgkadri98**

### CODE REVIEW

File: configs/config.yaml [hunk 1]
```diff
@@ -11,3 +11,7 @@ KNOWLEDGE_DB_ENDPOINT: "http://localhost:3500"
 ACS_ENDPOINT: ""
 ACS_API_KEY: ""
 ACS_API_VERSION: ""
+
+# Pyflowkit configuration
+PYFLOWKIT_ENDPOINT: "http://localhost:50052"
+PYFLOWKIT_API_KEY: "flowkit-python-api-key"
```
Comment (RobPasMue) [inline configs/config.yaml:17]: Shouldn't it be FLOWKIT_PYTHON rather than PYFLOWKIT?

**Review by RobPasMue — APPROVED** [review]
Too many lines of code for reviewing easily. I would recommend we start doing smaller PRs for the sake of the reviewers... I just skimmed through the content - LGTM

File: pkg/externalfunctions/dataextraction.go [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+//
+// Returns:
+//   - githubFilesToExtract: github files to extract.
+func DataExtractionGetGithubFilesToExtract(githubRepoName string, githubRepoOwner string,
+	githubRepoBranch string, githubAccessToken string, githubFileExtensions []string,
+	githubFilteredDirectories []string, githubExcludedDirectories []string) (githubFilesToExtract []string) {
+
+	client, ctx := dataExtractNewGithubClient(githubAccessToken)
+
+	// Retrieve the specified branch's SHA (commit hash) from the GitHub repository. This is used to identify the latest state of the branch.
```
Comment (RobPasMue) [inline pkg/externalfunctions/dataextraction.go:42]:
```suggestion
	// Retrieve the specified branch SHA (commit hash) from the GitHub repository. This is used to identify the latest state of the branch.
```

Try committing this change... fingers crossed

**Review by RobPasMue — CHANGES_REQUESTED** [review]
These changes are not working @Revathyvenugopal162 - the artifact does not contain the information now...

**Review by FelixKuhnAnsys — APPROVED** [review]

**Pull request merged**